src/cigar.py

```diff
@@ -100,7 +100,7 @@ def _summation(cont_list):
     temp = []
     cigar = []
     for i,l in enumerate(edits):
-        if l == edits[-1]:
+        if i == (len(edits)-1):
```

Comment:
If you want to treat the last element in a sequence differently, you don't check for it in every iteration of the loop. At least not in a fast language; so you can be excused in Python, I suppose. But generally, branches are expensive to have in tight loops. They are not always easy to avoid, and sometimes the code is _much_ simpler if we include them, but still, as an ideal, we try to avoid unnecessary cases in code we are going to execute a lot.

Even in Python, though, the code is harder to read if I have to keep two cases in mind when I'm reading the body of a loop, and it's always better to move special cases aside from the common case.

Instead, you loop through the first `n - 1` elements and do what you have to do there

```python
for i, l in enumerate(edits[:-1]):
    ...
```

(perhaps with a slightly better approach than slicing) and then you can deal with the last element in an `else` part of the loop. (The `else` block in Python loops is executed if you make it all the way through the loop without a `break`).